src/model.rs

```diff
@@ -119,6 +119,27 @@ pub struct Collections {
 }
 
 impl Collections {
+    /// Remove associated schedules with points of route
+    pub fn remove_points_of_route(&mut self) {
+        let vj_idxs: Vec<Idx<VehicleJourney>> =
+            self.vehicle_journeys.iter().map(|(idx, _)| idx).collect();
+        for vj_idx in vj_idxs {
+            let mut vj = self.vehicle_journeys.index_mut(vj_idx);
+
+            let is_point_of_route = |stop_time: &mut StopTime| -> bool {
+                stop_time.pickup_type == 3 || stop_time.drop_off_type == 3
+            };
```

Comment:
1. This function can be declared outside of the `for` loop (and might be named `is_route_point` if you change the other one too).
2. I don't think this closure needs a mutable reference, since it's only about reading information. A simple reference is probably enough.